src/instantx.cpp

```diff
@@ -372,6 +376,15 @@ bool ProcessConsensusVote(CNode* pnode, CConsensusVote& ctx)
                 if(pwalletMain){
                     if(pwalletMain->UpdatedTransaction((*i).second.txHash)){
                         nCompleteTXLocks++;
+                        if(nSignatures == INSTANTX_SIGNATURES_REQUIRED) {
+                            // notify external script once threshold is reached
+                            std::string strCmd = GetArg("-ixnotify", "");
+                            if ( !strCmd.empty())
+                            {
+                                boost::replace_all(strCmd, "%s", ctx.txHash.GetHex());
+                                boost::thread t(runCommand, strCmd); // thread runs free
```

Comment:
From experience, there's oddness if this runs from any other thread than init.cpp. Have you tested it? You should try sleeping for a minute in a target script, I think it will halt the daemon.